### PR TITLE
Fix more hardcoded paths

### LINKED ISSUE #14: Simplify the branching logic after inflation selection
### Description of feature

We do so much CSV filtering we probably don't need to or we can do it much more simply.

### PR DESCRIPTION
**Pull request by mertcelebi**
As I was trying to make the pipeline work with Tower, I found more hardcoded paths. This basically removes the reliance of using another CSV for transferring files over after the `FILTER_ORTHOGROUP` process.

Also fixes #14

### CODE REVIEW

File: modules/local/nf-core-modified/mafft.nf [hunk 1]
```diff
@@ -1,5 +1,8 @@
+// This process is originall from nf-core but is modified to remove the
```
Comment (mertcelebi) [inline modules/local/nf-core-modified/mafft.nf:1]:
```suggestion
// This process is originally from nf-core but is modified to remove the
```

File: modules/local/filter_orthogroups.nf [hunk 2]
```diff
@@ -52,17 +54,21 @@ process FILTER_ORTHOGROUPS {
...
+    while IFS= read -r line
+    do
+        # Copy the file to the destination directory
+        cp "\$line" SpeciesTreeFastas/
+    done < spptree_core_og_fpaths.txt
+    while IFS= read -r line
+    do
+        # Copy the file to the destination directory
+        cp "\$line" GeneTreeFastas/
+    done < genetree_core_og_fpaths.txt
```
Comment (taylorreiter) [inline modules/local/filter_orthogroups.nf:72]: can this not be done with the publishDir directive?

File: modules/local/nf-core-modified/mafft.nf [hunk 3]
```diff
@@ -9,26 +12,27 @@ process MAFFT {
...
+    prefix=\$(basename "${fasta}" .fa)
+
     mafft \\
         --thread ${task.cpus} \\
         --localpair \\
         --maxiterate 1000 \\
         --anysymbol \\
         ${fasta} \\
-        > ${prefix}-mafft.fa
+        > \$prefix-mafft.fa
```
Comment (taylorreiter) [inline modules/local/nf-core-modified/mafft.nf:35]: why do the curly braces break here? or rather why did you remove them?

**Review by taylorreiter — COMMENTED** [review]
I don't totally understand the changes to workflows/phylorthology.nf so I'm just commenting with on this PR. Let me know if it's time sensitive and I can try and dive deeper and approve.

File: modules/local/nf-core-modified/mafft.nf (hunk 3 above)
Comment (mertcelebi) [reply modules/local/nf-core-modified/mafft.nf:35]: One refers to a bash variable (new), the other refers to a groovy variable (old)

File: modules/local/filter_orthogroups.nf (hunk 2 above)
Comment (mertcelebi) [reply modules/local/filter_orthogroups.nf:72]: Nope, `publishDir` doesn't determine what gets uploaded, it determines where things get uploaded to

File: modules/local/clipkit.nf [hunk 4]
```diff
@@ -12,10 +12,10 @@ process CLIPKIT {
     )
 
     input:
-    tuple val(meta), path(fasta)   // Filepaths to the MSAs
+    path(fasta)   // Filepaths to the MSAs
```
Comment (austinhpatton) [inline modules/local/clipkit.nf:15]: So you did end up just getting away from our usage of the tuple here - I think that probably makes good sense, since I don't believe we need to use any of the associated metadata here (or need to retain them in the output)

File: modules/local/clipkit.nf [hunk 5]
```diff
@@ -25,8 +25,10 @@ process CLIPKIT {
     script:
     def args = task.ext.args ?: ''
     """
+    prefix=\$(basename "${fasta}" -mafft.fa)
```
Comment (austinhpatton) [inline modules/local/clipkit.nf:28]: I see, so by just stripping the rest of the filename from the MSA you're accomplishing the same thing as if you used the orthogroup name from the meta map (originally provided in a tuple). will be a lot cleaner.

File: modules/local/nf-core-modified/iqtree.nf [hunk 6]
```diff
@@ -9,13 +9,13 @@ process IQTREE {
         'quay.io/biocontainers/iqtree:2.1.4_beta--hdcc8f71_0' }"
 
     input:
-    tuple val(meta), path(alignment)
```
Comment (austinhpatton) [inline modules/local/nf-core-modified/iqtree.nf:12]: yeahhhh, probably good to just distance ourselves from using tuples like this whenever possible.

File: workflows/phylorthology.nf [hunk 7]
```diff
@@ -64,30 +66,12 @@ include { GENERAX                                   } from '../modules/local/gen
...
 //
-include { MAFFT                                 } from '../modules/nf-core/mafft/main' // Slightly modified (specific parameters)
-include { MAFFT as MAFFT_REMAINING              } from '../modules/nf-core/mafft/main'
 
 /*
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     RUN MAIN WORKFLOW
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 */
-// Function to get list of [ meta, [ file ] ]
```
Comment (austinhpatton) [inline workflows/phylorthology.nf:75]: yeah, I'm glad you were able to get rid of this - felt a bit hacky/extraneous, and again, not having to deal with the tuples for all those downstream modules will be a big improvement.

File: workflows/phylorthology.nf [hunk 8]
```diff
@@ -229,36 +213,16 @@ workflow PHYLORTHOLOGY {
...
-        .set { ch_rem_ogs }
-
     //
     // MODULE: MAFFT
     // Infer multiple sequence alignments of orthogroups/gene
     // families using MAFFT
     //
     // For the extreme core set to be used in species tree inference
-    ch_core_og_msas = MAFFT(ch_core_ogs).msas
+    ch_core_og_msas = MAFFT(FILTER_ORTHOGROUPS.out.spptree_fas.flatten()).msas
```
Comment (austinhpatton) [inline workflows/phylorthology.nf:222]: Much cleaner way of dealing with this.

File: workflows/phylorthology.nf [hunk 9]
```diff
@@ -279,50 +243,6 @@ workflow PHYLORTHOLOGY {
...
-
-    ch_rem_trimmed_msas
-    .branch {
-        meta, trimmed_msas ->
-            msas  : trimmed_msas
-                return trimmed_msas
-    }
-    .collect()
-    .set { ch_all_rem_msas }
-
```
Comment (austinhpatton) [inline workflows/phylorthology.nf:325]: SO happy to have this gone. Was just ugly, and I always had a feeling that making all of those branches was actually useless/accomplished nothing?

**Review by austinhpatton — APPROVED** [review]
This all looks great! So happy to have gotten away from the use of the tuples in those later modules - I think I originally used that approach because I wasn't sure how else to switch from distributing replicate tasks across species to across orthogroups, and so used the 'samplesheet' strategy used in the beginning since that's what i was familiar with. This makes things immensely cleaner though, and I'm very happy to have all that clutter removed from the main workflow.
